### ISSUE TITLE
`xkeyboard` module `%icon%` variable in `<label-layout>` token empty on startup

### ISSUE THREAD

**Issue by misopeth** [polybar/polybar#1906]
At startup the `%icon%` variable in the `label-layout` token is empty.
The `xkeyboard` module is configured like this:
```dosini
[module/keyboard]
type = internal/xkeyboard
format = <label-layout>
label-layout = %{T2}KBD%{T-} %icon%
layout-icon-default =
layout-icon-0 = it;IT
layout-icon-1 = gb;UK
indicator-icon-default =
indicator-icon-0 = caps lock;-CL;+CL
indicator-icon-1 = scroll lock;-SL;+SL
indicator-icon-2 = num lock;-NL;+NL
label-indicator-on = %icon%
label-indicator-off = %icon%
```
so at startup I expect to see `**KBD** IT`, but instead I only see `**KBD**`. The variable stays empty also when changing the keyboard layout. For reference, this is the content of my `/etc/X11/xorg.conf.d/00-keyboard.conf`:
```
Section "InputClass"
    Identifier "system-keyboard"
    MatchIsKeyboard "on"
    Option "XkbLayout" "it,gb"
    Option "XkbModel" "pc105"
    Option "XkbOptions" "ctrl:nocaps,grp:shifts_toggle,terminate:ctrl_alt_bksp"
EndSection
```
I'm on Arch, startup sequence is: LightDM, i3, polybar.

If I kill and start polybar again then everything works fine.

**Comment by patrick96** [comment]
So this only happens when you start polybar during boot?

How and where exactly do you start polybar?

Could you post the polybar output when this happens? (You may need to redirect its output to a logfile).

What are the values of `%name%` and `%layout%` when this happens?

**Comment by misopeth** [comment]
Yes. I start polybar through my i3 config file, using this line:
```
exec_always ~/.config/polybar/launch.sh
```
where `launch.sh` is this script:
```
#!/usr/bin/env bash

# Terminate already running bar instances
killall -q polybar

# Wait until the processes have been shut down
while pgrep -u $UID -x polybar >/dev/null; do sleep 1; done

# Launch bars
for mon in $(polybar -m | cut -d":" -f1); do
    MONITOR=$mon polybar i3 &>/home/fra/Downloads/polybar.log &
done
```

This is the content of the polybar log:
```
polybar|warn:  pulseaudio: using default sink alsa_output.pci-0000_00_1f.3.analog-stereo
polybar|error: Failed to query _XEMBED_INFO, removing client... (XCB_WINDOW (3))
```

When I have this problem `%layout%` is `us`, but the keyboard behaves correctly (i.e. it has the it layout).

**Comment by misopeth** [comment]
If I log out and log back in (instead of, for example, rebooting), the problem doesn't show up, and `%layout%` is correctly set to `it`.

**Comment by michaelweidmann** [comment]
Hi! I have got a similar problem. 

At startup the `%name%` variable contains the value `English (US)` although I have got a German keyboard layout. So I am expecting the value `German`. When switching to the russian keyboard layout, the variable is empty. Although the variable is wrong, the layouts are correct. After reloading the bar the variable contains for both layouts the right value.

Here is the configuration of the module:
```ini
[module/xkeyboard]
type = internal/xkeyboard
blacklist-0 = num lock
blacklist-1 = scroll lock

format-prefix = " "
format-prefix-foreground = ${colors.foreground-alt}
format-prefix-underline = ${colors.secondary}

label-layout = %name:0:3%
label-layout-underline = ${colors.secondary}

label-indicator-padding = 2
label-indicator-margin = 1
label-indicator-background = ${colors.secondary}
label-indicator-underline = ${colors.secondary}
```

My keyboard layout configuration in X11 looks like this:
```
Section "InputClass"
        Identifier "system-keyboard"
        MatchIsKeyboard "on"
        Option "XkbLayout" "de,de"
        Option "XkbModel" "pc105"
        Option "XkbVariant" "nodeadkeys,ru"
	Option "XkbOptions" "grp:ctrls_toggle"
EndSection
```
I am also on Arch with i3 and polybar but I don't have LigthDM installed. My polybar starts just like @misopeth's.

**Comment by sgtxn** [comment]
Same problem, using Arch with bspwm.  
```
➜ cat /etc/X11/xorg.conf.d/00-keyboard.conf
Section "InputClass"
        Identifier "system-keyboard"
        MatchIsKeyboard "on"
        Option "XkbLayout" "us,ru"
        Option "XkbModel" "pc104"
        Option "XkbOptions" "grp:caps_toggle,terminate:ctrl_alt_bksp"
EndSection
```
 
Although I've noticed that executing this line during startup  
```setxkbmap -layout us,ru &```  
seems to fix the issue for me. Though it still might be a coincidence, since it _sometimes_ worked fine even before I added this.

**Comment by misopeth** [comment]
> Although I've noticed that executing this line during startup seems to fix the issue for me.
> `setxkbmap -layout us,ru &`

This solves it also for me.

> Though it still might be a coincidence, since it sometimes worked fine even before I added this.

I guess it's a matter of who gets execute first, if polybar or the Xorg keyboard configuration, so I don't know if this can be fixed "within" polybar.

**Comment by sgtxn** [comment]
> I guess it's a matter of who gets execute first

That's what I also thought and tried adding ```sleep 2``` to launch.sh in order to delay polybar startup for 2, then 5 and even 20 seconds. Sadly, none of those helped and I still got an empty layout token, which was then instantly fixed by the aforementioned ```setxkbmap -layout us,ru``` while already running polybar.

**Comment by patrick96** [comment]
There is a bug in the XCB library (which we use for all interactions with the X server) where initially, the keyboard layout is set to `us`: https://github.com/polybar/polybar/wiki/Known-Issues#xkeyboard-module-starts-with-us-layout-at-startup

At least for me, this goes away after pressing any key, but evidently it doesn't for you.

There are two possibilities:

1. The xkeyboard module doesn't properly listen to new events and thus never updates the layout.
2. No such event is produced

The fact that just running `setxkbmap -layout us,ru` seems to fix it, suggests that polybar does properly listen to events, but it just never receives any that tell it that the layout isn't actually `us`.

I can unfortunately not reproduce this, so I can't really dig into this any further.
